src/components/entity/state-badge.ts

```diff
@@ -117,6 +121,12 @@ export class StateBadge extends LitElement {
           iconStyle.filter = `brightness(${(brightness + 245) / 5}%)`;
         }
       }
+      if (Array.isArray(this.stateColor)) {
+        const color = computeCustomStateColor(stateObj.state, this.stateColor);
+        if (color) {
+          iconStyle.color = color;
```

Comment:
Do you need to add an `else` clause to make sure you unassign `color` if you go form array -> boolean.

Reply:
There's existing logic that handles `state_color: true` that is unaffected by this, if I'm understanding your questions correctly (I'm not sure that I am)